assets/src/containers/App.js

```diff
@@ -1,64 +1,33 @@
 import React, { useState } from 'react'
-import { BrowserRouter as Router, Route } from 'react-router-dom'
-import DashboardAppBar from './DashboardAppBar'
-import SideDrawer from './SideDrawer'
-import GradeDistribution from './GradeDistribution'
-import AssignmentPlanning from './AssignmentPlanning'
-import FilesAccessed from './FilesAccessed'
-import IndexPage from './IndexPage'
-import Spinner from '../components/Spinner'
-import Error from './Error'
-import { isObjectEmpty } from '../util/object'
-import { useCourseInfo } from '../service/api'
+import { Route, withRouter } from 'react-router-dom'
+import { matchPath } from 'react-router';
+import GoogleAnalyticsTracking from '../components/GoogleAnalyticsTracking'
+import CourseList from './CourseList'
+import Course from './Course'
 
 function App (props) {
-  const { match } = props
-  const courseId = match.params.courseId
-  const [loaded, error, courseInfo] = useCourseInfo(courseId)
-  const [sideDrawerState, setSideDrawerState] = useState(false)
+  const {
+    location
+  } = props
+  const isLoggedIn = !!myla_globals.username
+  // replace with redirect to react login page later
+  if (!isLoggedIn) {
+    return (window.location.href = myla_globals.login)
+  }
 
-  // this is temporary
+  const coursePageMatch = matchPath(location.pathname, '/courses/:courseId/')
+  const courseId = coursePageMatch ? coursePageMatch.params.courseId : null
```

Comment:
This is a nice refactoring.

Reply:
I feel there should be separate URLs for courses list page `http://dev.edu/courses` the base URL which is `http://dev.edu`. Currently, both are tracked under `/`

Reply:
for now I've added another route (`/courses`) that also shows the CourseList screen

Reply:
What I mean is `CourseList` page should only be used for `/courses`

Reply:
done

Reply:
We still need `<Route path='/' exact render={props => <CourseList {...props} user={user} />} />` since after login it is redirecting to no component. This is fine for now. We need to change later

Reply:
I've added `<Route path='/' exact render={props => <CourseList {...props} user={user} />} />` back in